Add coolix climate ❄ 🔥

glmnet opened the pull request:
## Description:
Climate Coolix Infrared remote controller.
I am pushing this one alone, I'll definitely do two more (Tcl and Whirlpool from IRRemoteESP8266) but doing this alone first so I learn more on code styles, get the code reviewed, etc.

Tested functions 
  * OFF / AUTO / COOL, setting temperature from 17 to 30
  * accepts **optional** temperature sensor to display in thermostat card

Sample use:
````yaml
sensor:
  - platform: dht
    pin: D2
    model: DHT22
    temperature:
      name: "Estudio Test Temperature"
      id: estudio_test_dht_temp
    humidity:
      name: "Estudio Test Humidity"
    update_interval: 5s

remote_transmitter:
  carrier_duty_percent: 50%
  pin: D5

climate:
  platform: coolix
  name: 'AC Estudio'
  #sensor: estudio_test_dht_temp
````

**Related issue (if applicable):** 
feature: https://github.com/esphome/feature-requests/issues/24
old PR: https://github.com/esphome/esphome/pull/518

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>


## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

glmnet committed:
Lint

glmnet committed:
add coolix climate ❄ 🔥

glmnet committed:
Fixes

glmnet commented:
Having a hard time with linting.
Can sb tell me which tools should I install locally so this does not fail late on travis? I'm using vscode

timmo001 commented:
> Having a hard time with linting.
> Can sb tell me which tools should I install locally so this does not fail late on travis? I'm using vscode

Since the build is failing on cpplint I'd suggest installing the cpplint extension. Also, make sure to use the formatters out there

glmnet commented:
@timmo001 thanks, there are some many results when I search cpp lint that I was lost, now it fixes some of the issues but there are others which are not catch and also complains about stuff that travis does not.
Is there like a settings file I have to provide to cpplint?

OttoWinter reviewed (commented):
The formatter is clang-format (v7). Though that can be kinda hard to install - hence the travis script.

Probably the issue here is trailing whitespace, the suggest-changes script could be updated to highlight those better (there must be a git diff option for that).

From a quick look, this PR looks good 👍 Will take a closer look on the weekend.

glmnet committed:
Reviewed

OttoWinter reviewed (approved):
Looks good to me! (and sorry for the delay btw, totally missed the reminder I set to review this PR, sorry)

Do you think you can get the docs for this ready soon? I'd love to have this feature in 1.13 :P

glmnet commented:
Yes, i just updated both climate yesterday with code changes, comments in
one applies to the other.
I'll do the docs, I was hopping to work on top of base climate docs and
peek on bang bang, are these docs ready? Did not check docs repos lately.

On Wed, May 15, 2019 at 4:06 PM Otto Winter <notifications@github.com>
wrote:

> *@OttoWinter* approved this pull request.
>
> Looks good to me! (and sorry for the delay btw, totally missed the
> reminder I set to review this PR, sorry)
>
> Do you think you can get the docs for this ready soon? I'd love to have
> this feature in 1.13 :P
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/esphome/esphome/pull/521?email_source=notifications&email_token=AAAXPIXSHV2PTNOEDZ7TCIDPVRNLJA5CNFSM4HIJECNKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOBYX5EJY#pullrequestreview-238015015>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAXPIROEH62ATP3IQJ5SADPVRNLJANCNFSM4HIJECNA>
> .
>


-- 
*GLM*

OttoWinter commented:
> I'll do the docs, I was hopping to work on top of base climate docs and peek on bang bang, are these docs ready? 

Yep, see https://next.esphome.io (`next` branch in esphome-docs repo)

OttoWinter committed:
Merge branch 'dev' into coolix-climate

OttoWinter committed:
Fix for dev

ClimateDevice was renamed to Climate

OttoWinter committed:
Remove stale method

OttoWinter committed:
Lint

OttoWinter committed:
Initialize target temperature, avoid NAN value

OttoWinter committed:
Use clamp and round value

OttoWinter committed:
Set to verbose message

Not really relevant to the user

OttoWinter committed:
Remove constructor

Name is now set in climate.register_climate - saves integrations from having to declare a constructor

OttoWinter committed:
Fix, add test

OttoWinter committed:
Merge branch 'dev' into coolix-climate

OttoWinter committed:
Merge branch 'dev' into coolix-climate

# Conflicts:
#	tests/test1.yaml
